tests/ecosystem/upgrade/test_logging_upgrade.py

```diff
@@ -0,0 +1,165 @@
+
+        """
+
+        ceph_cluster = CephCluster()
+        with CephHealthMonitor(ceph_cluster):
+
+            #  Pre-check
+            check_cluster_before_after_upgrade()
+
+            # Matching the OCP version and cluster-Logging version
+            ocp_version = get_ocp_version()
+            logger.info(f"OCP version {ocp_version}")
+            csv = ocp.OCP(
+                kind=constants.CLUSTER_SERVICE_VERSION,
+                namespace=constants.OPENSHIFT_LOGGING_NAMESPACE
+            )
+            csv_info = csv.get(
+                out_yaml_format=True
+            )
+            clo_version_before_upgrade = (
+                csv_info['items'][0]['spec']['version']
+            )
+            eso_version_before_upgrade = (
+                csv_info['items'][1]['spec']['version']
+            )
+            logger.info(
+                f"Cluster-logging operator version before upgrade "
+                f"{clo_version_before_upgrade}"
+            )
+            logger.info(
+                f"Elastic search operator version after upgrade "
+                f"{eso_version_before_upgrade}"
+            )
+            subscription = get_clusterlogging_subscription()
+            logging_channel = subscription['spec']['channel']
+            logger.info(
+                f"Logging channel before upgrade {logging_channel}"
+            )
+            upgrade_channel = config.UPGRADE['upgrade_logging_channel']
+            if ocp_version > logging_channel:
+                # Upgrade Elastic search operator Subscription
+                es_subscription_cmd = (
+                    f'oc patch subscription elasticsearch-operator '
+                    f'-n {constants.OPENSHIFT_OPERATORS_REDHAT_NAMESPACE} '
+                    f'--type merge -p \'{{"spec":{{"channel": '
+                    f'"{upgrade_channel}"}}}}\''
+                )
+                # Upgrade Cluster-logging operator subscription
+                clo_subscription_cmd = (
+                    f'oc patch subscription cluster-logging '
+                    f'-n {constants.OPENSHIFT_LOGGING_NAMESPACE} '
+                    f'--type merge -p \'{{"spec":{{"channel": '
+                    f'"{upgrade_channel}"}}}}\''
+                )
+                run_cmd(es_subscription_cmd)
+                run_cmd(clo_subscription_cmd)
+                for sampler in TimeoutSampler(
+                    timeout=600,
+                    sleep=60,
+                    func=check_csv_logging_upgrade_phase
+                ):
+                    logger.info(
+                        f"Logging upgrade "
+                        f"{'completed!' if sampler else 'did not completed yet!'}"
+                    )
+                    if sampler:
+                        break
+            else:
+                logger.info(
+                    "Logging Version matches the OCP version, No upgrade needed"
+                )
+
+            check_cluster_before_after_upgrade()
```

Comment:
Add a check to see the versions of the logging are correct after the upgrade.

Reply:
Added